nominations/models.py

```diff
@@ -355,6 +470,76 @@ def save(self, *args, **kwargs):
         if self.vol_notes and not self.staff_notes:
             self.staff_notes = self.vol_notes
 
+    def create_related_objects(self):
+        """Create related nomination and questionnaire for application."""
+        if not self.nomination:
+            self.nomination = Nomination.objects.create()
+
+        if not self.questionnaire:
+            self.questionnaire = Questionnaire.objects.create()
+
+    def generate_application_status(self):
+        """Returns a generated status based on completion of various items."""
+
+        if self.nomination.status == 'incomplete':
+            if self.questionnaire.status == 'complete':
+                status = 'needs-group-form'
+            else:
+                status = 'needs-group-form-and-questionnaire'
+        else:
+            # nomination complete
+            if self.questionnaire.status == 'incomplete':
+                # needs questionaire
+                status = 'needs-questionnaire'
+            else:
+                # questionnaire complete
+                status = 'incomplete'
+        return status
+
+    def is_editable(self):
+        """Returns whether the user can edit this application."""
+
+        # TODO possibly move this to settings.py or env
+        editable_statuses = [
+            'needs-group-form-and-questionnaire',
+            'needs-questionnaire',
+            'needs-group-form',
+            'incomplete'
+        ]
+
+        if self.status in editable_statuses:
+            return True
+        else:
+            return False
+
+    def __unicode__(self):
+        return str(self.group) + ' - ' + self.candidate_first_name + ' ' + self.candidate_last_name
+
+    def save(self, *args, **kwargs):
+        if not self.nomination or not self.questionnaire:
+            self.create_related_objects()
+
+        self.auto_populate_research_fields()
+
+        # TODO move to settings.py or env?
+        # dont run status generator if set to these
+        exempt_statuses = [
```

Comment:
Is this just the inverse of editable status list?

Also in case like this I suggest to use whitelist instead of blacklist. If we create a new status and don't update the logic here, what should we expect to be default behavior? Probably would prefer to treat as exempt by default? Although I suggest just move this logic into the status getter method anyway and just set the status every time.

Reply:
It is just inverse, there was a bug with just using inverse but I'll work through it since it is what should be done.